source/ada/lsp-ada_documents.adb

```diff
@@ -2854,13 +2855,23 @@ package body LSP.Ada_Documents is
       Filter.Initialize (Token, Node);
 
       for Provider of Providers loop
-         Provider.Propose_Completion
-           (Sloc   => Sloc,
-            Token  => Token,
-            Node   => Node,
-            Filter => Filter,
-            Names  => Names,
-            Result => Result);
+         begin
+            Provider.Propose_Completion
+              (Sloc   => Sloc,
+               Token  => Token,
+               Node   => Node,
+               Filter => Filter,
+               Names  => Names,
+               Result => Result);
+
+         exception
+            when E : others =>
```

Comment:
should we purge Result/Names or save them before and restore them in case of error?

Reply:
No, because the exception handler is here to allow having results even in an exception happened (e,g: when a minor completion provider had an exception we want to have the results of all the providers that have succeed).

Reply:
fine, the exception handler is a bit to wide: could we reduce it to Property_Error?